src/core/toga/images.py

```diff
@@ -8,14 +8,22 @@ class Image:
     :param path: Path to the image. Allowed values can be local file
         (relative or absolute path) or URL (HTTP or HTTPS). Relative paths
         will be interpreted relative to the application module directory.
+        TODO: or bytes object containig a valid image...
     """
-    def __init__(self, path):
-        if isinstance(path, pathlib.Path):
-            self.path = path
-        elif path.startswith('http://') or path.startswith('https://'):
-            self.path = path
+    def __init__(self, value=None, path=None):
+        if path is not None:
+            value = path  # TODO: code to deprecate path
```

Comment:
I can see what you're doing here, but the error handling on this is incomplete. It allows specifying both `value` and `path`, and doesn't catch the `Image()` case (at least, not before crashing later).

I'd also suggest changing the function prototype to:

    def __init__(self, value=None, *, path=None, data=None):

If you specify `value`, you get full "autodetection" behavior; if you specify `path`, you get the existing "path or url" conversion; and if you specify `data`, it's assumed to be in the right format. Specifying more than one argument would be an error. This doesn't have to be a deprecation path, either - it doesn't seem unreasonable as an ongoing API.

Reply:
I'd never encountered that lonely star in the args before. Powerful stuff.

If your preference is to preserve separate `path` and `data` arguments, I personally would prefer this function prototype:
```
def __init__(self, path=None, data=None):
```

Simpler, more obvious. One way of doing things. More Zen.

But you're the BDFL. Let me know.

Reply:
That's a good point - I got so focussed on the change introducing `value` that I missed the obvious option. Let's go with `(path=None, data=None)`.

FYI - there's an analogous `/` that matches `*` in args. `*` forces all subsequent arguments to be kwargs; `/` forces all prior arguments to be positional. The full form of a function is effectively:

    def name(positional_only_parameters, /, positional_or_keyword_parameters, *, keyword_only_parameters):

See [PEP570](https://peps.python.org/pep-0570/) (and [PEP457](https://peps.python.org/pep-0457/)) for details.

Reply:
In that case, `data` should probably be a keyword-only argument, while `path` can continue to be either keyword or positional, for backward compatibility. In other words:
```
def __init__(self, path=None, *, data=None):
```